src/wallet_client.ts

```diff
@@ -174,32 +174,36 @@ export class WalletClient {
   async transferSuiMnemonic(
     amount: number,
     suiAccount: Ed25519Keypair,
-    receiverAddress: SuiAddress
+    receiverAddress: SuiAddress,
+    typeArg: string = SUI_TYPE_ARG
   ) {
     const keypair = suiAccount;
     const senderAddress = keypair.getPublicKey().toSuiAddress();
     const coinsNeeded =
       await this.provider.selectCoinSetWithCombinedBalanceGreaterThanOrEqual(
         senderAddress,
-        BigInt(amount + DEFAULT_GAS_BUDGET_FOR_SUI_TRANSFER)
+        BigInt(amount),
```

Comment:
we need to allocate gas budget

Reply:
A seperate function call is used in L189 for gas object

Reply:
Added a if else for sui trasactions

Reply:
Currently using 1000 as default gas because it was what sui initially was using